src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -138,9 +137,16 @@ private SecurityAnalysisResult runSimulations(LfNetwork network, List<Propagated
                 }
 
                 // start a simulation for each of the contingency
-                Iterator<LfContingency> contingencyIt = contingencies.iterator();
+                Iterator<PropagatedContingency> contingencyIt = propagatedContingencies.iterator();
```

Comment:
Aside this PR, but I'm using it to ask: do we need a iterator? Shouldn't we restore the pre-contingency state at the end?

Reply:
This is an optimization because after last contingency simulation, we don't need to restore base case state.
No idea if it worse it or not...